File: tmdb-core/src/main/java/dev/reuss/tmdb/common/keyword/Keyword.java
package dev.reuss.tmdb.common.keyword;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * TMDB keyword.
 *
 * @param id   TMDB keyword id
 * @param name keyword name
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record Keyword(
        int id,
        String name
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/common/keyword/Keyword.kt
package dev.reuss.tmdb.common.keyword

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * TMDB keyword.
 *
 * @property id   TMDB keyword id
 * @property name keyword name
 */
@JvmRecord
data class Keyword(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("name")
    val name: String?,
) : TmdbModel
